### ISSUE TITLE
com.onesignal.SynService ANR crash

### ISSUE THREAD

**Issue by iballan** [OneSignal/OneSignal-Android-SDK#246]
This crash happens on Android 6.0 and 7.0

from the log i saw that it happens in:

```
at com.onesignal.ActivityLifecycleHandler.onActivityResumed (ActivityLifecycleHandler.java:74)
  at com.onesignal.ActivityLifecycleListener.onActivityResumed (ActivityLifecycleListener.java:50)

```

This is the full log:

```

"Signal Catcher" daemon prio=5 tid=2 Runnable
  | group="system" sCount=0 dsCount=0 obj=0x12f640a0 self=0x7f76951000
  | sysTid=22043 nice=0 cgrp=default sched=0/0 handle=0x7f7c03f440
  | state=R schedstat=( 3880938 438907 12 ) utm=0 stm=0 core=5 HZ=100
  | stack=0x7f7bf43000-0x7f7bf45000 stackSize=1013KB
  | held mutexes= "mutator lock"(shared held)
  native: pc 000000000048a858  /system/lib64/libart.so (art::DumpNativeStack(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, int, char const*, art::ArtMethod*, void*)+236)
  native: pc 0000000000459e80  /system/lib64/libart.so (art::Thread::Dump(std::__1::basic_ostream<char, std::__1::char_traits<char> >&) const+220)
  native: pc 00000000004665f8  /system/lib64/libart.so (art::DumpCheckpoint::Run(art::Thread*)+688)
  native: pc 0000000000467510  /system/lib64/libart.so (art::ThreadList::RunCheckpoint(art::Closure*)+276)
  native: pc 0000000000467bcc  /system/lib64/libart.so (art::ThreadList::Dump(std::__1::basic_ostream<char, std::__1::char_traits<char> >&)+188)
  native: pc 00000000004684b4  /system/lib64/libart.so (art::ThreadList::DumpForSigQuit(std::__1::basic_ostream<char, std::__1::char_traits<char> >&)+492)
  native: pc 0000000000431c84  /system/lib64/libart.so (art::Runtime::DumpForSigQuit(std::__1::basic_ostream<char, std::__1::char_traits<char> >&)+96)
  native: pc 000000000043f308  /system/lib64/libart.so (art::SignalCatcher::HandleSigQuit()+1256)
  native: pc 000000000043ff7c  /system/lib64/libart.so (art::SignalCatcher::Run(void*)+552)
  native: pc 00000000000660c4  /system/lib64/libc.so (__pthread_start(void*)+52)
  native: pc 000000000001ee84  /system/lib64/libc.so (__start_thread+16)

"main" prio=5 tid=1 Blocked
  | group="main" sCount=1 dsCount=0 obj=0x77204f70 self=0x7f7d0bba00
  | sysTid=22038 nice=0 cgrp=default sched=0/0 handle=0x7f807d4fc8
  | state=S schedstat=( 201493739 27593022 255 ) utm=15 stm=5 core=4 HZ=100
  | stack=0x7fdd8c4000-0x7fdd8c6000 stackSize=8MB
  | held mutexes=
  at com.onesignal.LocationGMS.fireComplete (LocationGMS.java:208)
- waiting to lock <0x05c51372> (a java.lang.Class<com.onesignal.i>) held by thread 22 (tid=22)
  at com.onesignal.LocationGMS.fireFailedComplete (LocationGMS.java:202)
  at com.onesignal.LocationGMS.getLocation (LocationGMS.java:152)
  at com.onesignal.OneSignal.startLocationUpdate (OneSignal.java:441)
  at com.onesignal.OneSignal.startRegistrationOrOnSession (OneSignal.java:434)
  at com.onesignal.OneSignal.onAppFocus (OneSignal.java:712)
  at com.onesignal.ActivityLifecycleHandler.handleFocus (ActivityLifecycleHandler.java:120)
  at com.onesignal.ActivityLifecycleHandler.onActivityResumed (ActivityLifecycleHandler.java:74)
  at com.onesignal.ActivityLifecycleListener.onActivityResumed (ActivityLifecycleListener.java:50)
  at android.app.Application.dispatchActivityResumed (Application.java:214)
  at android.app.Activity.onResume (Activity.java:1212)
  at android.support.v4.app.FragmentActivity.onResume (FragmentActivity.java:485)
  at com.mbh.namazvakitleri.activities.BaseActivity.onResume (BaseActivity.java:158)
  at android.app.Instrumentation.callActivityOnResume (Instrumentation.java:1259)
  at android.app.Activity.performResume (Activity.java:6347)
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:3110)
  at android.app.ActivityThread.handleResumeActivity (ActivityThread.java:3152)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2499)
  at android.app.ActivityThread.access$900 (ActivityThread.java:157)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1356)
  at android.os.Handler.dispatchMessage (Handler.java:102)
  at android.os.Looper.loop (Looper.java:148)
  at android.app.ActivityThread.main (ActivityThread.java:5551)
  at java.lang.reflect.Method.invoke! (Native method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:730)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:620)

"ReferenceQueueDaemon" daemon prio=5 tid=3 Waiting
  | group="system" sCount=1 dsCount=0 obj=0x12f609a0 self=0x7f6e18ca00
  | sysTid=22044 nice=0 cgrp=default sched=0/0 handle=0x7f7bf40440
  | state=S schedstat=( 910048 374271 14 ) utm=0 stm=0 core=5 HZ=100
  | stack=0x7f7be3e000-0x7f7be40000 stackSize=1037KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x0d0233c3> (a java.lang.Class<java.lang.ref.ReferenceQueue>)
  at java.lang.Daemons$ReferenceQueueDaemon.run (Daemons.java:147)
- locked <0x0d0233c3> (a java.lang.Class<java.lang.ref.ReferenceQueue>)
  at java.lang.Thread.run (Thread.java:818)

"FinalizerDaemon" daemon prio=5 tid=4 Waiting
  | group="system" sCount=1 dsCount=0 obj=0x12f60a00 self=0x7f6e18de00
  | sysTid=22045 nice=0 cgrp=default sched=0/0 handle=0x7f7be3b440
  | state=S schedstat=( 1926719 976821 17 ) utm=0 stm=0 core=5 HZ=100
  | stack=0x7f7bd39000-0x7f7bd3b000 stackSize=1037KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x0187a440> (a java.lang.ref.ReferenceQueue)
  at java.lang.Object.wait (Object.java:423)
  at java.lang.ref.ReferenceQueue.remove (ReferenceQueue.java:101)
- locked <0x0187a440> (a java.lang.ref.ReferenceQueue)
  at java.lang.ref.ReferenceQueue.remove (ReferenceQueue.java:72)
  at java.lang.Daemons$FinalizerDaemon.run (Daemons.java:185)
  at java.lang.Thread.run (Thread.java:818)

"FinalizerWatchdogDaemon" daemon prio=5 tid=5 Waiting
  | group="system" sCount=1 dsCount=0 obj=0x12f60a60 self=0x7f6e18e800
  | sysTid=22046 nice=0 cgrp=default sched=0/0 handle=0x7f7bd36440
  | state=S schedstat=( 1126875 3926509 20 ) utm=0 stm=0 core=0 HZ=100
  | stack=0x7f7bc34000-0x7f7bc36000 stackSize=1037KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x044c9f79> (a java.lang.Daemons$FinalizerWatchdogDaemon)
  at java.lang.Daemons$FinalizerWatchdogDaemon.waitForObject (Daemons.java:255)
- locked <0x044c9f79> (a java.lang.Daemons$FinalizerWatchdogDaemon)
  at java.lang.Daemons$FinalizerWatchdogDaemon.run (Daemons.java:227)
  at java.lang.Thread.run (Thread.java:818)

"HeapTaskDaemon" daemon prio=5 tid=6 Blocked
  | group="system" sCount=1 dsCount=0 obj=0x12f60ac0 self=0x7f6e18f200
  | sysTid=22047 nice=0 cgrp=default sched=0/0 handle=0x7f7bc31440
  | state=S schedstat=( 42978227 4728125 55 ) utm=0 stm=4 core=3 HZ=100
  | stack=0x7f7bb2f000-0x7f7bb31000 stackSize=1037KB
  | held mutexes=
  native: pc 000000000001c210  /system/lib64/libc.so (syscall+32)
  native: pc 00000000001391b8  /system/lib64/libart.so (art::ConditionVariable::TimedWait(art::Thread*, long, int)+164)
  native: pc 0000000000268060  /system/lib64/libart.so (art::gc::TaskProcessor::GetTask(art::Thread*)+228)
  native: pc 0000000000268664  /system/lib64/libart.so (art::gc::TaskProcessor::RunAllTasks(art::Thread*)+120)
  native: pc 000000000001375c  /system/framework/arm64/boot.oat (Java_dalvik_system_VMRuntime_runHeapTasks__+128)
  at dalvik.system.VMRuntime.runHeapTasks (Native method)
- waiting to lock an unknown object
  at java.lang.Daemons$HeapTaskDaemon.run (Daemons.java:355)
  at java.lang.Thread.run (Thread.java:818)

"Binder_1" prio=5 tid=7 Native
  | group="main" sCount=1 dsCount=0 obj=0x12f6b0a0 self=0x7f6e18c000
  | sysTid=22048 nice=0 cgrp=default sched=0/0 handle=0x7f7ba2e440
  | state=S schedstat=( 3501038 3741982 27 ) utm=0 stm=0 core=0 HZ=100
  | stack=0x7f7b932000-0x7f7b934000 stackSize=1013KB
  | held mutexes=
  native: pc 0000000000068c88  /system/lib64/libc.so (__ioctl+4)
  native: pc 0000000000072670  /system/lib64/libc.so (ioctl+100)
  native: pc 000000000002d3b4  /system/lib64/libbinder.so (android::IPCThreadState::talkWithDriver(bool)+164)
  native: pc 000000000002dc88  /system/lib64/libbinder.so (android::IPCThreadState::getAndExecuteCommand()+24)
  native: pc 000000000002dda4  /system/lib64/libbinder.so (android::IPCThreadState::joinThreadPool(bool)+76)
  native: pc 00000000000367e8  /system/lib64/libbinder.so (???)
  native: pc 0000000000015954  /system/lib64/libutils.so (android::Thread::_threadLoop(void*)+208)
  native: pc 000000000009392c  /system/lib64/libandroid_runtime.so (android::AndroidRuntime::javaThreadShell(void*)+96)
  native: pc 00000000000151a4  /system/lib64/libutils.so (???)
  native: pc 00000000000660c4  /system/lib64/libc.so (__pthread_start(void*)+52)
  native: pc 000000000001ee84  /system/lib64/libc.so (__start_thread+16)

"Binder_2" prio=5 tid=8 Native
  | group="main" sCount=1 dsCount=0 obj=0x12f9e0a0 self=0x7f76957400
  | sysTid=22049 nice=0 cgrp=default sched=0/0 handle=0x7f7b92f440
  | state=S schedstat=( 3091875 6286929 20 ) utm=0 stm=0 core=5 HZ=100
  | stack=0x7f7b833000-0x7f7b835000 stackSize=1013KB
  | held mutexes=
  native: pc 0000000000068c88  /system/lib64/libc.so (__ioctl+4)
  native: pc 0000000000072670  /system/lib64/libc.so (ioctl+100)
  native: pc 000000000002d3b4  /system/lib64/libbinder.so (android::IPCThreadState::talkWithDriver(bool)+164)
  native: pc 000000000002dc88  /system/lib64/libbinder.so (android::IPCThreadState::getAndExecuteCommand()+24)
  native: pc 000000000002dda4  /system/lib64/libbinder.so (android::IPCThreadState::joinThreadPool(bool)+76)
  native: pc 00000000000367e8  /system/lib64/libbinder.so (???)
  native: pc 0000000000015954  /system/lib64/libutils.so (android::Thread::_threadLoop(void*)+208)
  native: pc 000000000009392c  /system/lib64/libandroid_runtime.so (android::AndroidRuntime::javaThreadShell(void*)+96)
  native: pc 00000000000151a4  /system/lib64/libutils.so (???)
  native: pc 00000000000660c4  /system/lib64/libc.so (__pthread_start(void*)+52)
  native: pc 000000000001ee84  /system/lib64/libc.so (__start_thread+16)

"Queue" prio=10 tid=9 Waiting
  | group="main" sCount=1 dsCount=0 obj=0x12f98be0 self=0x7f6e190600
  | sysTid=22051 nice=-8 cgrp=default sched=0/0 handle=0x7f7b738440
  | state=S schedstat=( 10468701 2135414 40 ) utm=1 stm=0 core=5 HZ=100
  | stack=0x7f7b636000-0x7f7b638000 stackSize=1037KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x0dccf4be> (a java.lang.Object)
  at java.lang.Thread.parkFor$ (Thread.java:1220)
- locked <0x0dccf4be> (a java.lang.Object)
  at sun.misc.Unsafe.park (Unsafe.java:299)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:158)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await (AbstractQueuedSynchronizer.java:2013)
  at java.util.concurrent.PriorityBlockingQueue.take (PriorityBlockingQueue.java:510)
  at io.fabric.sdk.android.services.concurrency.c.a (DependencyPriorityBlockingQueue.java:197)
  at io.fabric.sdk.android.services.concurrency.c.b (DependencyPriorityBlockingQueue.java:236)
  at io.fabric.sdk.android.services.concurrency.c.a (DependencyPriorityBlockingQueue.java:65)
  at io.fabric.sdk.android.services.concurrency.c.take (DependencyPriorityBlockingQueue.java:46)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1038)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1098)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:588)
  at java.lang.Thread.run (Thread.java:818)

"Queue" prio=10 tid=10 Waiting
  | group="main" sCount=1 dsCount=0 obj=0x12f98c40 self=0x7f7d0bc400
  | sysTid=22052 nice=10 cgrp=bg_non_interactive sched=0/0 handle=0x7f7b633440
  | state=S schedstat=( 107745774 13821725 172 ) utm=9 stm=1 core=5 HZ=100
  | stack=0x7f7b531000-0x7f7b533000 stackSize=1037KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x0bca491f> (a java.lang.Object)
  at java.lang.Thread.parkFor$ (Thread.java:1220)
- locked <0x0bca491f> (a java.lang.Object)
  at sun.misc.Unsafe.park (Unsafe.java:299)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:158)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await (AbstractQueuedSynchronizer.java:2013)
  at java.util.concurrent.PriorityBlockingQueue.take (PriorityBlockingQueue.java:510)
  at io.fabric.sdk.android.services.concurrency.c.a (DependencyPriorityBlockingQueue.java:197)
  at io.fabric.sdk.android.services.concurrency.c.b (DependencyPriorityBlockingQueue.java:236)
  at io.fabric.sdk.android.services.concurrency.c.a (DependencyPriorityBlockingQueue.java:65)
  at io.fabric.sdk.android.services.concurrency.c.take (DependencyPriorityBlockingQueue.java:46)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1038)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1098)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:588)
  at java.lang.Thread.run (Thread.java:818)

"Queue" prio=10 tid=11 Waiting
  | group="main" sCount=1 dsCount=0 obj=0x12f98ca0 self=0x7f7d0bce00
  | sysTid=22053 nice=10 cgrp=bg_non_interactive sched=0/0 handle=0x7f7b52c440
  | state=S schedstat=( 3169743 3077864 22 ) utm=0 stm=0 core=4 HZ=100
  | stack=0x7f7b42a000-0x7f7b42c000 stackSize=1037KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x0462c46c> (a java.lang.Object)
  at java.lang.Thread.parkFor$ (Thread.java:1220)
- locked <0x0462c46c> (a java.lang.Object)
  at sun.misc.Unsafe.park (Unsafe.java:299)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:158)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await (AbstractQueuedSynchronizer.java:2013)
  at java.util.concurrent.PriorityBlockingQueue.take (PriorityBlockingQueue.java:510)
  at io.fabric.sdk.android.services.concurrency.c.a (DependencyPriorityBlockingQueue.java:197)
  at io.fabric.sdk.android.services.concurrency.c.b (DependencyPriorityBlockingQueue.java:236)
  at io.fabric.sdk.android.services.concurrency.c.a (DependencyPriorityBlockingQueue.java:65)
  at io.fabric.sdk.android.services.concurrency.c.take (DependencyPriorityBlockingQueue.java:46)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1038)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1098)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:588)
  at java.lang.Thread.run (Thread.java:818)

"Queue" prio=10 tid=12 Waiting
  | group="main" sCount=1 dsCount=0 obj=0x12f98d00 self=0x7f7d0bd800
  | sysTid=22054 nice=10 cgrp=bg_non_interactive sched=0/0 handle=0x7f7b41f440
  | state=S schedstat=( 757970 594379 13 ) utm=0 stm=0 core=4 HZ=100
  | stack=0x7f7b31d000-0x7f7b31f000 stackSize=1037KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x0d6eba35> (a java.lang.Object)
  at java.lang.Thread.parkFor$ (Thread.java:1220)
- locked <0x0d6eba35> (a java.lang.Object)
  at sun.misc.Unsafe.park (Unsafe.java:299)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:158)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await (AbstractQueuedSynchronizer.java:2013)
  at java.util.concurrent.PriorityBlockingQueue.take (PriorityBlockingQueue.java:510)
  at io.fabric.sdk.android.services.concurrency.c.a (DependencyPriorityBlockingQueue.java:197)
  at io.fabric.sdk.android.services.concurrency.c.b (DependencyPriorityBlockingQueue.java:236)
  at io.fabric.sdk.android.services.concurrency.c.a (DependencyPriorityBlockingQueue.java:65)
  at io.fabric.sdk.android.services.concurrency.c.take (DependencyPriorityBlockingQueue.java:46)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1038)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1098)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:588)
  at java.lang.Thread.run (Thread.java:818)

"Queue" prio=10 tid=13 Waiting
  | group="main" sCount=1 dsCount=0 obj=0x12f98d60 self=0x7f6e18fc00
  | sysTid=22055 nice=-8 cgrp=default sched=0/0 handle=0x7f76b81440
  | state=S schedstat=( 515260 434895 8 ) utm=0 stm=0 core=4 HZ=100
  | stack=0x7f76a7f000-0x7f76a81000 stackSize=1037KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x0edf9eca> (a java.lang.Object)
  at java.lang.Thread.parkFor$ (Thread.java:1220)
- locked <0x0edf9eca> (a java.lang.Object)
  at sun.misc.Unsafe.park (Unsafe.java:299)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:158)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await (AbstractQueuedSynchronizer.java:2013)
  at java.util.concurrent.PriorityBlockingQueue.take (PriorityBlockingQueue.java:510)
  at io.fabric.sdk.android.services.concurrency.c.a (DependencyPriorityBlockingQueue.java:197)
  at io.fabric.sdk.android.services.concurrency.c.b (DependencyPriorityBlockingQueue.java:236)
  at io.fabric.sdk.android.services.concurrency.c.a (DependencyPriorityBlockingQueue.java:65)
  at io.fabric.sdk.android.services.concurrency.c.take (DependencyPriorityBlockingQueue.java:46)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1038)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1098)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:588)
  at java.lang.Thread.run (Thread.java:818)

"Queue" prio=10 tid=14 Waiting
  | group="main" sCount=1 dsCount=0 obj=0x12f98dc0 self=0x7f7c076200
  | sysTid=22056 nice=-8 cgrp=default sched=0/0 handle=0x7f7693f440
  | state=S schedstat=( 626666 741614 13 ) utm=0 stm=0 core=5 HZ=100
  | stack=0x7f7683d000-0x7f7683f000 stackSize=1037KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x01d4583b> (a java.lang.Object)
  at java.lang.Thread.parkFor$ (Thread.java:1220)
- locked <0x01d4583b> (a java.lang.Object)
  at sun.misc.Unsafe.park (Unsafe.java:299)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:158)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await (AbstractQueuedSynchronizer.java:2013)
  at java.util.concurrent.PriorityBlockingQueue.take (PriorityBlockingQueue.java:510)
  at io.fabric.sdk.android.services.concurrency.c.a (DependencyPriorityBlockingQueue.java:197)
  at io.fabric.sdk.android.services.concurrency.c.b (DependencyPriorityBlockingQueue.java:236)
  at io.fabric.sdk.android.services.concurrency.c.a (DependencyPriorityBlockingQueue.java:65)
  at io.fabric.sdk.android.services.concurrency.c.take (DependencyPriorityBlockingQueue.java:46)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1038)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1098)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:588)
  at java.lang.Thread.run (Thread.java:818)

"Queue" prio=10 tid=15 Waiting
  | group="main" sCount=1 dsCount=0 obj=0x12f98e20 self=0x7f7c076c00
  | sysTid=22057 nice=10 cgrp=bg_non_interactive sched=0/0 handle=0x7f7683a440
  | state=S schedstat=( 799949 1381199 10 ) utm=0 stm=0 core=4 HZ=100
  | stack=0x7f76738000-0x7f7673a000 stackSize=1037KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x02edbb58> (a java.lang.Object)
  at java.lang.Thread.parkFor$ (Thread.java:1220)
- locked <0x02edbb58> (a java.lang.Object)
  at sun.misc.Unsafe.park (Unsafe.java:299)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:158)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await (AbstractQueuedSynchronizer.java:2013)
  at java.util.concurrent.PriorityBlockingQueue.take (PriorityBlockingQueue.java:510)
  at io.fabric.sdk.android.services.concurrency.c.a (DependencyPriorityBlockingQueue.java:197)
  at io.fabric.sdk.android.services.concurrency.c.b (DependencyPriorityBlockingQueue.java:236)
  at io.fabric.sdk.android.services.concurrency.c.a (DependencyPriorityBlockingQueue.java:65)
  at io.fabric.sdk.android.services.concurrency.c.take (DependencyPriorityBlockingQueue.java:46)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1038)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1098)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:588)
  at java.lang.Thread.run (Thread.java:818)

"Answers Events Handler1" prio=5 tid=16 Native
  | group="main" sCount=1 dsCount=0 obj=0x13027880 self=0x7f7b7dea00
  | sysTid=22060 nice=10 cgrp=bg_non_interactive sched=0/0 handle=0x7f61d71440
  | state=S schedstat=( 84356936 15183586 139 ) utm=7 stm=1 core=3 HZ=100
  | stack=0x7f61c6f000-0x7f61c71000 stackSize=1037KB
  | held mutexes=
  native: pc 0000000000068a94  /system/lib64/libc.so (__pselect6+8)
  native: pc 0000000000022fc4  /system/lib64/libc.so (select+140)
  native: pc 0000000000011180  /system/lib64/libjavacrypto.so (???)
  native: pc 000000000001a2a0  /system/lib64/libjavacrypto.so (???)
  native: pc 0000000000fc6d48  /system/framework/arm64/boot.oat (Java_com_android_org_conscrypt_NativeCrypto_SSL_1do_1handshake__JLjava_io_FileDescriptor_2Lcom_android_org_conscrypt_NativeCrypto_00024SSLHandshakeCallbacks_2IZ_3B_3B+268)
  at com.android.org.conscrypt.NativeCrypto.SSL_do_handshake (Native method)
  at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake (OpenSSLSocketImpl.java:324)
  at com.android.okhttp.internal.http.SocketConnector.connectTls (SocketConnector.java:123)
  at com.android.okhttp.Connection.connect (Connection.java:151)
  at com.android.okhttp.Connection.connect (Connection.java:187)
  at com.android.okhttp.Connection.connectAndSetOwner (Connection.java:238)
  at com.android.okhttp.OkHttpClient$1.connectAndSetOwner (OkHttpClient.java:134)
  at com.android.okhttp.internal.http.HttpEngine.nextConnection (HttpEngine.java:382)
  at com.android.okhttp.internal.http.HttpEngine.connect (HttpEngine.java:369)
  at com.android.okhttp.internal.http.HttpEngine.sendRequest (HttpEngine.java:287)
  at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute (HttpURLConnectionImpl.java:457)
  at com.android.okhttp.internal.huc.HttpURLConnectionImpl.connect (HttpURLConnectionImpl.java:130)
  at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getOutputStream (HttpURLConnectionImpl.java:261)
  at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getOutputStream (DelegatingHttpsURLConnection.java:218)
  at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getOutputStream (HttpsURLConnectionImpl.java)
  at io.fabric.sdk.android.services.network.HttpRequest.m (HttpRequest.java:2450)
  at io.fabric.sdk.android.services.network.HttpRequest.n (HttpRequest.java:2463)
  at io.fabric.sdk.android.services.network.HttpRequest.a (HttpRequest.java:2655)
  at io.fabric.sdk.android.services.network.HttpRequest.a (HttpRequest.java:2617)
  at com.crashlytics.android.answers.SessionAnalyticsFilesSender.send (SessionAnalyticsFilesSender.java:45)
  at com.crashlytics.android.answers.AnswersRetryFilesSender.send (AnswersRetryFilesSender.java:49)
  at com.crashlytics.android.answers.EnabledSessionAnalyticsManagerStrategy.sendEvents (EnabledSessionAnalyticsManagerStrategy.java:154)
  at com.crashlytics.android.answers.AnswersEventsHandler$3.run (AnswersEventsHandler.java:103)
  at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:423)
  at java.util.concurrent.FutureTask.run (FutureTask.java:237)
  at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run (ScheduledThreadPoolExecutor.java:269)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1113)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:588)
  at jk$1$1.onRun (ExecutorUtils.java:75)
  at je.run (BackgroundPriorityRunnable.java:30)
  at java.lang.Thread.run (Thread.java:818)

"Crashlytics Exception Handler1" prio=5 tid=18 Waiting
  | group="main" sCount=1 dsCount=0 obj=0x13114520 self=0x7f7b7dfe00
  | sysTid=22065 nice=10 cgrp=bg_non_interactive sched=0/0 handle=0x7f61c6c440
  | state=S schedstat=( 16649422 6083698 43 ) utm=1 stm=0 core=5 HZ=100
  | stack=0x7f61b6a000-0x7f61b6c000 stackSize=1037KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x0c5dc4b1> (a java.lang.Object)
  at java.lang.Thread.parkFor$ (Thread.java:1220)
- locked <0x0c5dc4b1> (a java.lang.Object)
  at sun.misc.Unsafe.park (Unsafe.java:299)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:158)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await (AbstractQueuedSynchronizer.java:2013)
  at java.util.concurrent.LinkedBlockingQueue.take (LinkedBlockingQueue.java:410)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1038)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1098)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:588)
  at jk$1$1.onRun (ExecutorUtils.java:75)
  at je.run (BackgroundPriorityRunnable.java:30)
  at java.lang.Thread.run (Thread.java:818)

"FocusHandlerThread" prio=5 tid=19 Native
  | group="main" sCount=1 dsCount=0 obj=0x1310cd60 self=0x7f61b0e800
  | sysTid=22070 nice=0 cgrp=default sched=0/0 handle=0x7f6172a440
  | state=S schedstat=( 483646 5966093 14 ) utm=0 stm=0 core=4 HZ=100
  | stack=0x7f61628000-0x7f6162a000 stackSize=1037KB
  | held mutexes=
  native: pc 00000000000693f4  /system/lib64/libc.so (__epoll_pwait+8)
  native: pc 000000000001f4e4  /system/lib64/libc.so (epoll_pwait+32)
  native: pc 000000000001af2c  /system/lib64/libutils.so (android::Looper::pollInner(int)+144)
  native: pc 000000000001b30c  /system/lib64/libutils.so (android::Looper::pollOnce(int, int*, int*, void**)+80)
  native: pc 00000000000d6298  /system/lib64/libandroid_runtime.so (android::NativeMessageQueue::pollOnce(_JNIEnv*, _jobject*, int)+48)
  native: pc 00000000008e8f7c  /system/framework/arm64/boot.oat (Java_android_os_MessageQueue_nativePollOnce__JI+144)
  at android.os.MessageQueue.nativePollOnce (Native method)
  at android.os.MessageQueue.next (MessageQueue.java:323)
  at android.os.Looper.loop (Looper.java:135)
  at android.os.HandlerThread.run (HandlerThread.java:61)

"OSH_NetworkHandlerThread" prio=5 tid=20 Native
  | group="main" sCount=1 dsCount=0 obj=0x131bb0b0 self=0x7f61b0f200
  | sysTid=22071 nice=0 cgrp=default sched=0/0 handle=0x7f61625440
  | state=S schedstat=( 1389324 0 5 ) utm=0 stm=0 core=5 HZ=100
  | stack=0x7f61523000-0x7f61525000 stackSize=1037KB
  | held mutexes=
  native: pc 00000000000693f4  /system/lib64/libc.so (__epoll_pwait+8)
  native: pc 000000000001f4e4  /system/lib64/libc.so (epoll_pwait+32)
  native: pc 000000000001af2c  /system/lib64/libutils.so (android::Looper::pollInner(int)+144)
  native: pc 000000000001b30c  /system/lib64/libutils.so (android::Looper::pollOnce(int, int*, int*, void**)+80)
  native: pc 00000000000d6298  /system/lib64/libandroid_runtime.so (android::NativeMessageQueue::pollOnce(_JNIEnv*, _jobject*, int)+48)
  native: pc 00000000008e8f7c  /system/framework/arm64/boot.oat (Java_android_os_MessageQueue_nativePollOnce__JI+144)
  at android.os.MessageQueue.nativePollOnce (Native method)
  at android.os.MessageQueue.next (MessageQueue.java:323)
  at android.os.Looper.loop (Looper.java:135)
  at android.os.HandlerThread.run (HandlerThread.java:61)

"AsyncTask #1" prio=5 tid=21 Waiting
  | group="main" sCount=1 dsCount=0 obj=0x131c71c0 self=0x7f7b7e1200
  | sysTid=22074 nice=10 cgrp=bg_non_interactive sched=0/0 handle=0x7f61520440
  | state=S schedstat=( 950362 623440 8 ) utm=0 stm=0 core=5 HZ=100
  | stack=0x7f6141e000-0x7f61420000 stackSize=1037KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x07c0dd96> (a java.lang.Object)
  at java.lang.Thread.parkFor$ (Thread.java:1220)
- locked <0x07c0dd96> (a java.lang.Object)
  at sun.misc.Unsafe.park (Unsafe.java:299)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:158)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await (AbstractQueuedSynchronizer.java:2013)
  at java.util.concurrent.LinkedBlockingQueue.take (LinkedBlockingQueue.java:410)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1038)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1098)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:588)
  at java.lang.Thread.run (Thread.java:818)

"AsyncTask #2" prio=5 tid=23 Waiting
  | group="main" sCount=1 dsCount=0 obj=0x131ce160 self=0x7f61b05c00
  | sysTid=22076 nice=10 cgrp=bg_non_interactive sched=0/0 handle=0x7f61316440
  | state=S schedstat=( 770206 477554 15 ) utm=0 stm=0 core=5 HZ=100
  | stack=0x7f61214000-0x7f61216000 stackSize=1037KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x08a63d17> (a java.lang.Object)
  at java.lang.Thread.parkFor$ (Thread.java:1220)
- locked <0x08a63d17> (a java.lang.Object)
  at sun.misc.Unsafe.park (Unsafe.java:299)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:158)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await (AbstractQueuedSynchronizer.java:2013)
  at java.util.concurrent.LinkedBlockingQueue.take (LinkedBlockingQueue.java:410)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1038)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1098)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:588)
  at java.lang.Thread.run (Thread.java:818)

"OS_SYNCSRV_BG_SYNC" prio=5 tid=22 TimedWaiting
  | group="main" sCount=1 dsCount=0 obj=0x131c75e0 self=0x7f61b02a00
  | sysTid=22079 nice=0 cgrp=default sched=0/0 handle=0x7f6141b440
  | state=S schedstat=( 4607344 9495624 18 ) utm=0 stm=0 core=5 HZ=100
  | stack=0x7f61319000-0x7f6131b000 stackSize=1037KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x0c2f3504> (a java.lang.Object)
  at java.lang.Thread.join (Thread.java:786)
- locked <0x0c2f3504> (a java.lang.Object)
  at java.lang.Thread.join (Thread.java:744)
  at com.onesignal.ac.c (OneSignalRestClient.java:102)
  at com.onesignal.ac.d (OneSignalRestClient.java:86)
  at com.onesignal.z.a (OneSignal.java:696)
  at com.onesignal.SyncService.b (SyncService.java:53)
  at com.onesignal.SyncService.a (SyncService.java:42)
  at com.onesignal.SyncService$2$1.a (SyncService.java:92)
  at com.onesignal.i.a (LocationGMS.java:208)
- locked <0x05c51372> (a java.lang.Class<com.onesignal.i>)
  at com.onesignal.i.b (LocationGMS.java:202)
  at com.onesignal.i.a (LocationGMS.java:152)
  at com.onesignal.SyncService$2.run (SyncService.java:85)
  at java.lang.Thread.run (Thread.java:818)

"OS_HTTPConnection" prio=5 tid=24 Native
  | group="main" sCount=1 dsCount=0 obj=0x131ec820 self=0x7f76963200
  | sysTid=22080 nice=0 cgrp=default sched=0/0 handle=0x7f61211440
  | state=S schedstat=( 4324637 3520781 12 ) utm=0 stm=0 core=0 HZ=100
  | stack=0x7f6110f000-0x7f61111000 stackSize=1037KB
  | held mutexes=
  native: pc 0000000000068a94  /system/lib64/libc.so (__pselect6+8)
  native: pc 0000000000022f94  /system/lib64/libc.so (select+92)
  native: pc 0000000000011180  /system/lib64/libjavacrypto.so (???)
  native: pc 000000000001a2a0  /system/lib64/libjavacrypto.so (???)
  native: pc 0000000000fc6d48  /system/framework/arm64/boot.oat (Java_com_android_org_conscrypt_NativeCrypto_SSL_1do_1handshake__JLjava_io_FileDescriptor_2Lcom_android_org_conscrypt_NativeCrypto_00024SSLHandshakeCallbacks_2IZ_3B_3B+268)
  at com.android.org.conscrypt.NativeCrypto.SSL_do_handshake (Native method)
  at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake (OpenSSLSocketImpl.java:324)
  at com.android.okhttp.internal.http.SocketConnector.connectTls (SocketConnector.java:123)
  at com.android.okhttp.Connection.connect (Connection.java:151)
  at com.android.okhttp.Connection.connect (Connection.java:187)
  at com.android.okhttp.Connection.connectAndSetOwner (Connection.java:238)
  at com.android.okhttp.OkHttpClient$1.connectAndSetOwner (OkHttpClient.java:134)
  at com.android.okhttp.internal.http.HttpEngine.nextConnection (HttpEngine.java:382)
  at com.android.okhttp.internal.http.HttpEngine.connect (HttpEngine.java:369)
  at com.android.okhttp.internal.http.HttpEngine.sendRequest (HttpEngine.java:287)
  at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute (HttpURLConnectionImpl.java:457)
  at com.android.okhttp.internal.huc.HttpURLConnectionImpl.connect (HttpURLConnectionImpl.java:130)
  at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getOutputStream (HttpURLConnectionImpl.java:261)
  at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getOutputStream (DelegatingHttpsURLConnection.java:218)
  at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getOutputStream (HttpsURLConnectionImpl.java)
  at com.onesignal.ac.d (OneSignalRestClient.java:142)
  at com.onesignal.ac.b (OneSignalRestClient.java:38)
  at com.onesignal.ac$4.run (OneSignalRestClient.java:94)
  at java.lang.Thread.run (Thread.java:818)

"pool-3-thread-1" prio=5 tid=17 Waiting
  | group="main" sCount=1 dsCount=0 obj=0x12f64460 self=0x7f61b05200
  | sysTid=22095 nice=0 cgrp=default sched=0/0 handle=0x7f60b40440
  | state=S schedstat=( 2120470 1584532 16 ) utm=0 stm=0 core=4 HZ=100
  | stack=0x7f60a3e000-0x7f60a40000 stackSize=1037KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x0758faed> (a java.lang.Object)
  at java.lang.Thread.parkFor$ (Thread.java:1220)
- locked <0x0758faed> (a java.lang.Object)
  at sun.misc.Unsafe.park (Unsafe.java:299)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:158)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await (AbstractQueuedSynchronizer.java:2013)
  at java.util.concurrent.LinkedBlockingQueue.take (LinkedBlockingQueue.java:410)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1038)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1098)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:588)
  at java.lang.Thread.run (Thread.java:818)

"OkHttp ConnectionPool" daemon prio=10 tid=25 TimedWaiting
  | group="main" sCount=1 dsCount=0 obj=0x12f70a00 self=0x7f7b2acc00
  | sysTid=22096 nice=-8 cgrp=default sched=0/0 handle=0x7f619bf440
  | state=S schedstat=( 578959 99636 1 ) utm=0 stm=0 core=4 HZ=100
  | stack=0x7f618bd000-0x7f618bf000 stackSize=1037KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x0da53d22> (a com.android.okhttp.ConnectionPool)
  at com.android.okhttp.ConnectionPool.performCleanup (ConnectionPool.java:307)
- locked <0x0da53d22> (a com.android.okhttp.ConnectionPool)
  at com.android.okhttp.ConnectionPool.runCleanupUntilPoolIsEmpty (ConnectionPool.java:244)
  at com.android.okhttp.ConnectionPool.-wrap0 (ConnectionPool.java)
  at com.android.okhttp.ConnectionPool$1.run (ConnectionPool.java:98)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1113)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:588)
  at java.lang.Thread.run (Thread.java:818)

".namazvakitleri" prio=5 (not attached)
  | sysTid=22094 nice=0 cgrp=default
  | state=S schedstat=( 23506927 9851199 24 ) utm=2 stm=0 core=5 HZ=100

```

**Comment by jkasten2** [comment]
@iballan Sorry about this issue, the ANR is a duplicate of issue #237 and has been fixed in the 3.5.7 update.

Thanks.

**Comment by iballan** [comment]
it is happening in different places right ? all of them were fixed ?

**Comment by jkasten2** [comment]
@iballan They are from the same root cause of not setting up a thread handler with `GoogleApiClientListener`. Your stack trace is different as it is due to a failure case connecting instead of a success.
Thanks again still for posting this as it may help others if they each up searching for the same stack trace.

**Comment by iballan** [comment]
@jkasten2 oh got it.  Thank you for explaining :)

**Comment by VladD2** [comment]
>They are from the same root cause of not setting up a thread handler with GoogleApiClientListener. Your stack trace is different as it is due to a failure case connecting instead of a success.

Guys! We have a similar problem. It occurs on almost empty broadcast receivers. Explain how you fix it?

I did a search on GoogleApiClientListener, but did not find anything sensible.
